File: backend/app/models/email.py
from typing import List, Optional
from pydantic import BaseModel
from datetime import datetime


class EmailMessage(BaseModel):
    id: str
    subject: str
    body: str
    sender: str
    timestamp: datetime


class DraftReason(BaseModel):
    text: str
    confidence: float


class EmailDraft(BaseModel):
    id: str
    content: str
    reasons: List[DraftReason]


class FeedbackAction(BaseModel):
    action: str  # "accept", "reject", "edit", "ignore"
    reason: Optional[str] = None
    edited_content: Optional[str] = None


class UserFeedback(BaseModel):
    email_id: str
    draft_id: str
    action: FeedbackAction
    reason_ratings: dict[str, bool]  # reason_id -> like/dislike
    timestamp: datetime

File: frontend/src/types/email.ts
export interface EmailMessage {
  id: string;
  subject: string;
  body: string;
  sender: string;
  timestamp: string;
}

export interface DraftReason {
  text: string;
  confidence: number;
}

export interface EmailDraft {
  id: string;
  content: string;
  reasons: DraftReason[];
}

export interface FeedbackAction {
  action: 'accept' | 'reject' | 'edit' | 'ignore';
  reason?: string;
  edited_content?: string;
}

export interface UserFeedback {
  email_id: string;
  draft_id: string;
  action: FeedbackAction;
  reason_ratings: Record<string, boolean>;
  timestamp: string;
}